Src/FluentAssertions/Equivalency/Steps/GenericDictionaryEquivalencyStep.cs

```diff
@@ -39,71 +37,33 @@ public class GenericDictionaryEquivalencyStep : IEquivalencyStep
-        {
-            var (isDictionary, actualDictionary) = EnsureSubjectIsDictionary(comparands, expectedDictionary);
-
-            if (isDictionary && AssertSameLength(comparands, actualDictionary, expectedDictionary))
-            {
-                AssertDictionaryEquivalence(comparands, context, nestedValidator, actualDictionary, expectedDictionary);
-            }
-        }
-    }
-
     private static bool AssertSubjectIsNotNull(object subject)
     {
         return AssertionScope.Current
             .ForCondition(subject is not null)
             .FailWith("Expected {context:Subject} not to be {0}{reason}.", new object[] { null });
     }
 
-    private static bool AssertExpectationIsNotNull(object subject, object expectation)
-    {
-        return AssertionScope.Current
-            .ForCondition(expectation is not null)
-            .FailWith("Expected {context:Subject} to be {0}{reason}, but found {1}.", null, subject);
-    }
-
-    private static (bool isDictionary, DictionaryInterfaceInfo info) EnsureSubjectIsDictionary(Comparands comparands,
+    private static DictionaryInterfaceInfo EnsureSubjectIsDictionary(Comparands comparands,
         DictionaryInterfaceInfo expectedDictionary)
     {
-        bool isDictionary = DictionaryInterfaceInfo.TryGetFromWithKey(comparands.Subject.GetType(), "subject",
-            expectedDictionary.Key, out var actualDictionary);
+        var actualDictionary = DictionaryInterfaceInfo.FindFromWithKey(comparands.Subject.GetType(), "subject",
+            expectedDictionary.Key);
 
-        if (!isDictionary && expectedDictionary.TryConvertFrom(comparands.Subject, out var convertedSubject))
+        if (actualDictionary is null && expectedDictionary.ConvertFrom(comparands.Subject) is { } convertedSubject)
         {
             comparands.Subject = convertedSubject;
-            isDictionary = DictionaryInterfaceInfo.TryGetFrom(comparands.Subject.GetType(), "subject", out actualDictionary);
+            actualDictionary = DictionaryInterfaceInfo.FindFrom(comparands.Subject.GetType(), "subject");
         }
 
-        if (!isDictionary)
+        if (actualDictionary is null)
         {
             AssertionScope.Current.FailWith(
                 $"Expected {{context:subject}} to be a dictionary or collection of key-value pairs that is keyed to type {expectedDictionary.Key}. " +
                 $"It implements {actualDictionary}.");
         }
 
-        return (isDictionary, actualDictionary);
-    }
-
-    private static bool AssertSameLength(Comparands comparands, DictionaryInterfaceInfo actualDictionary,
-        DictionaryInterfaceInfo expectedDictionary)
-    {
-        if (comparands.Subject is ICollection subjectCollection
-            && comparands.Expectation is ICollection expectationCollection
-            && subjectCollection.Count == expectationCollection.Count)
-        {
-            return true;
-        }
```

Comment:
This is now done inside `AssertDictionaryEquivalence` where `comparands.Subject` and `comparands.Expectation` have been cast to `IDictionary<,>` so `Count` is directly available.